usbfreedom/core.py

```
@@ -227,13 +235,101 @@ def flash(self):
+        logger.info("Flashing image to boot partition...")
+        boot_partition = pm._get_partition_path(1)
+
+        cmd = [
+            'dd',
+            f'if={self.image_path}',
+            f'of={boot_partition}',
+            'bs=4M',
+            'status=progress',
+            'conv=fsync'
+        ]
+        subprocess.run(cmd, check=True)
```

Ruff (0.14.5)

`subprocess` call: check for execution of untrusted input

(S603)